doc: Replace m2r2 with myst-parser (#30259)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

IDE, OS, and tool support
=========================

> * Replaced ``m2r2`` with ``myst-parser`` in the documentation build requirements.

Reviewer: ```suggestion
* Updated documentation build requirements to replace ``m2r2`` with ``myst-parser``.
```
